### PR TITLE
:sparkles: Update to Kubernetes 1.12.2

### LINKED ISSUE #211: Update to k8s 1.12
k8s 1.12 has been out for some time now and 1.13 is [scheduled](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.13#kubernetes-113-release-schedule) for the start of December.

I'm taking a stab at updating the k8s deps but I wanted to check if there are any reservations on how this might affect kubebuilder or the controller-tools project e.g https://github.com/kubernetes-sigs/controller-runtime/pull/110. Or if there is an update process beyond simply updating the deps.

### PR DESCRIPTION
**Pull request by jimmidyson**
Fixes #211

This has no breaking changes, only changes to dependencies and internal changes to work with the new version.

### CODE REVIEW

**Comment by jimmidyson** [comment]
Anyone able to review this? I'd really like to see it merged ASAP.

**Comment by DirectXMan12** [comment]
I'll take a look -- last week was thanksgiving in the us, so I was out on break ;-)

**Comment by DirectXMan12** [comment]
Furthermore, you've confirmed that there are no changes to the few objects from the Kubernetes API that we expose in public interfaces?

**Comment by jimmidyson** [comment]
Hope Thanksgiving was a blast!

I'll verify tomorrow that there are no breaking changes to any interfaces this project exposes.

**Comment by jimmidyson** [comment]
@DirectXMan12 As far as I can tell, it should be backwards-compatible, but tbh I'm not sure I've looked at every single thing... Do you have any pointers of things to double/triple check? The fact that tests didn't need to change (other than more known types) gives me confidence that stuff hasn't broken.

**Comment by DirectXMan12** [comment]
I don't have anything at the moment.  Let me try and whip something up.

**Comment by jimmidyson** [comment]
Any more thoughts on this?

**Comment by DirectXMan12** [comment]
Finally finished enough of the tooling to run the check.  We're good in this regard.  I'll post the tooling soon, but I figured I'd give the results here and unblock this.

/lgtm
/approve

**Comment by DirectXMan12** [comment]
we can do the "latest patch release" in another PR if necessary.

File: pkg/client/unstructured_client.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func (uc *unstructuredClient) UpdateStatus(_ context.Context, obj runtime.Object
 	if err != nil {
 		return err
 	}
-	i, err := r.UpdateStatus(u)
+	i, err := r.UpdateStatus(u, metav1.UpdateOptions{})
```
Comment (DirectXMan12) [inline pkg/client/unstructured_client.go]: @droot we'll want to surface create options somewhere in our client at some point

File: pkg/manager/internal.go [hunk 2]
```diff
@@ -268,7 +268,7 @@ func (cm *controllerManager) startLeaderElection() (err error) {
 		RenewDeadline: 10 * time.Second,
 		RetryPeriod:   2 * time.Second,
 		Callbacks: leaderelection.LeaderCallbacks{
-			OnStartedLeading: func(_ <-chan struct{}) {
+			OnStartedLeading: func(_ context.Context) {
```
Comment (DirectXMan12) [inline pkg/manager/internal.go]: we'll want to surface this too

**Comment by DirectXMan12** [comment]
now we need to update controller-tools as well.

**Pull request merged**